src/Button/Button.less

```diff
@@ -22,7 +22,7 @@
 
 a, button {
   &.Button {
-    .type_medium;
+    .text--medium;
```

Comment:
slightly worried about making this a hard change - rather than leaving the old ones in first and then updating all codebases that are using them to the new one before deleting completely.

if they're being referenced only in LESS, it'll be fine since there will be build errors once people upgrade and they can fix them.
but if they're used as class names in JSX or jade/html, it'll be harder to detect the breaking change